assets/src/components/app.tsx

```diff
@@ -35,9 +35,9 @@ const App = (): JSX.Element => {
   const socket = useSocket()
   const vehiclesByRouteId = useVehicles(socket, selectedRouteIds)
 
-  const selectedRoutes = (routes || []).filter(route =>
-    selectedRouteIds.includes(route.id)
-  )
+  const selectedRoutes: Route[] = selectedRouteIds
+    .map(routeId => (routes || []).find(route => route.id === routeId))
```

Comment:
Extracting this out into a named function (like `findRouteById` or something) might help make this more readable.